### PR TITLE
Tao 874 test runner hook integration

### PR DESCRIPTION
**Pull request by ssipasseuth**
Adding a test runner client hook.
This is required for :
https://oat-sa.atlassian.net/browse/TAO-874
https://oat-sa.atlassian.net/browse/TAO-875

### CODE REVIEW

**Comment by krampstudio** [comment]
It looks OK, but the test is missing. `actionBarHook.js` can be tested without problems.

File: views/js/testRunner/actionBarHook.js [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+        var $toolsContainer = $('.tools-box-list');
+        var tplData = {
+            id : id,
+            navigation : false,
+            title : toolconfig.title || toolconfig.label,
+            label : toolconfig.label,
+            icon : toolconfig.icon
+        };
+        var $button = $(buttonTpl(tplData));
+        var amd = toolconfig.hook;
```
Comment (krampstudio) [inline views/js/testRunner/actionBarHook.js]: I'd like `toolconfig.hook` to be checked and/or adding a 3rd parameter to require to handle load errors

File: views/js/testRunner/actionBarHook.js [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+        
+        require([amd], function(hook){
+            if(_.isFunction(hook.init)){
```
Comment (krampstudio) [inline views/js/testRunner/actionBarHook.js]: Check also if hook is an object

File: views/js/testRunner/actionBarHook.js [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+                hook.init($button, toolconfig, assessmentTestContext);
+                
+                //only attach the button to the dom when everything is ready
+                $toolsContainer.append($button);
```
Comment (krampstudio) [inline views/js/testRunner/actionBarHook.js]: It would be useful to know when it's done (ie. trigger an event)

**Comment by ssipasseuth** [comment]
ok, I made the suggested changes, please review again

File: views/js/test/actionBarHook/hooks/validHook.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2015 (original work) Open Assessment Technologies SA ;
+ *
+ */
+define([], function(){
```
Comment (krampstudio) [inline views/js/test/actionBarHook/hooks/validHook.js]: just a hint (does not need update)
You can use a shorter syntax
```javascript```
define({
init : function init($btn, config, assessmentTestContext){ },
clear : function clear()
});
```

Comment (ssipasseuth) [inline views/js/test/actionBarHook/hooks/validHook.js]: ok, cool, thanks.

**Pull request merged**